daemons/attrd/attrd_ipc.c

```diff
@@ -481,6 +462,14 @@ attrd_ipc_dispatch(qb_ipcs_connection_t * c, void *data, size_t size)
             .result         = PCMK__UNKNOWN_RESULT,
         };
 
+        CRM_ASSERT(client->user != NULL);
+        pcmk__update_acl_user(xml, PCMK__XA_ATTR_USER, client->user);
+
+        if (client->name == NULL) {
+            const char *value = crm_element_value(xml, F_ORIG);
+            client->name = crm_strdup_printf("%s.%d", pcmk__s(value, "unknown"), client->pid);
+        }
```

Comment:
I suspect we don't need this now that we use (or should use) pcmk__client_name() everywhere

Reply:
It looks like the only places left referencing "client->name" are here, various places in mainloop.c that are unrelated, and several spots in the based and execd code which I do not think have been converted over yet.  So yeah this can go.

Reply:
Yeah only the attrd uses are relevant here